[Cisco Umbrella] adding support for Cisco Managed S3 buckets and non SQS

P1llus opened the pull request:
## What does this PR do?

Adds support for Non SQS type S3 buckets, including Cisco Managed S3 buckets.
Updates pipeline to support the newest proxy logs.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

elasticmachine commented:
Pinging @elastic/security-external-integrations (Team:Security-External Integrations)

P1llus commented:
I wish there was a better way to separate SQS and non SQS without being confusing about the settings :(

@kaiyan-sheng Tagging you as optional in case you also wanted to take a look, as it is S3 related.

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-2462/detail/PR-2462/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-2462/detail/PR-2462/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-2462/detail/PR-2462/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-2462/detail/PR-2462/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_2462.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2022-01-04T08:22:16.813Z&rangeTo=2022-01-04T08:42:16.813Z&transactionName=Ingest-manager%2Fintegrations%2FPR-%7Bnumber%7D&transactionType=unknown&latencyAggregationType=avg&traceId=b31becd4668b666dda5bb142d5ad21bb&transactionId=399916779a9e09ac)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2022-01-04T08:32:16.813+0000


* Duration: 13 min 32 sec


* Commit: 8d8534b6bffb0809f938db9aecfa163315d0db21



#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 7  |
| Skipped      | 0 |
| Total        | 7   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->












## :robot: GitHub comments

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




<!--COMMENT_GENERATED_WITH_ID_comment.id-->

aspacca commented on packages/cisco_umbrella/data_stream/log/manifest.yml:
```diff
@@ -11,9 +11,40 @@ streams:
+        required: false
         show_user: true
-        description: URL of the AWS SQS queue that messages will be received from.
+        description: Required for Cisco Managed S3. Number of workers that will process the S3 objects listed. Default is 1.
```
please, beware that this config param has no default (ie: default 0), if not set the underlying filebeat instance will exit. `Default is 1.` message might be misleading in this sense, not sure if it needs a better wording

aspacca replied:
missed https://github.com/elastic/integrations/blob/57a49493c3c3f0c89bbcda46ae4e879b65a4a08a/packages/cisco_umbrella/data_stream/log/manifest.yml#L47

aspacca commented:
@P1llus 
minor: https://github.com/elastic/beats/blob/master/x-pack/filebeat/input/awss3/config.go#L32 `bucket_list_interval` config param is missing, not sure if you want to add.

also having the s3 direct polling as input in the agent might be tricky, see: https://www.elastic.co/guide/en/beats/filebeat/7.15/filebeat-input-aws-s3.html#_parallel_processing

```
When using the polling list of S3 bucket objects method be aware that if running multiple Filebeat instances, they can list the same S3 bucket at the same time. Since the state of the ingested S3 objects is persisted (upon processing a single list operation) in the path.data configuration and multiple Filebeat cannot share the same path.data this will produce repeated ingestion of the S3 object. Therefore, when using the polling list of S3 bucket objects method, scaling should be vertical, with a single bigger Filebeat instance and higher number_of_workers config value.
```

Since with integrations the common case is to have the same policy dispatched to multiple agents will break due to the above behaviour.
I think we should add some big "warning" about creating a dedicated policy for the integration when dealing with Cisco Managed S3 buckets and linking it to a single agent.

P1llus commented:
Still adding some small fixes to the package

P1llus commented:
Had to add a few changes to make everything work as expected:

If you use non SQS, we want to ignore the auditlogs, as they are not in CSV format, or any recognized format:
```
{{#if bucket_list_prefix}}
file_selectors: 
  - regex: {{bucket_list_prefix}}/dnslogs/.+
  - regex: {{bucket_list_prefix}}/proxylogs/.+
  - regex: {{bucket_list_prefix}}/cloudfirewalllogs/.+
  - regex: {{bucket_list_prefix}}/iplogs/.+
{{/if}}
```
Moved away from the GROK pattern, and to the CSV format that the original pipeline used to make it more correct.
Added support for the new Proxy logs.

P1llus commented:
Package has been tested with a Cisco Managed S3 bucket and works correctly.

kaiyan-sheng commented on packages/cisco_umbrella/_dev/build/docs/README.md:
```diff
@@ -1,14 +1,16 @@
 
 ### Umbrella
 
+When using Cisco Managed S3 buckets that does not use SQS there is no load balancing possibilities for multiple agents, a single agent should be configured to poll the S3 bucket for new and updated files, and the number of workers can be configured to scale vertically
```
```suggestion
When using Cisco Managed S3 buckets that does not use SQS there is no load balancing possibilities for multiple agents, a single agent should be configured to poll the S3 bucket for new and updated files, and the number of workers can be configured to scale vertically.
```
